packages/core/src/App/Components/Layout/Footer/__tests__/toggle-language-settings.spec.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { useStore } from '@deriv/stores';
+import { ToggleLanguageSettings } from '../toggle-language-settings';
+import userEvent from '@testing-library/user-event';
+
+jest.mock('@deriv/stores', () => ({
+    ...jest.requireActual('@deriv/stores'),
+    useStore: jest.fn().mockReturnValue({
+        ui: {
+            is_language_settings_modal_on: false,
+            toggleLanguageSettingsModal: jest.fn(),
+        },
+        common: { current_language: 'EN' },
+    }),
+}));
```

Comment:
@yauheni-deriv Please use `mockStore` from `@deriv/stores` package instead, You need to wrap your component with `StoreProvider` for testing.